docs/design/proposed/interactive-mode.md

```diff
@@ -0,0 +1,64 @@
+
+### Detecting interactive mode
+
+* If the `--interactive` CLI flag was passed, then interactive mode should be used.
+* If the Console is not a TTY (i.e doesn't have an interactive terminal connected), then interactive mode should not be used. This can be detected via the following mechanisms
+  * if `System.Console.IsOutputRedirected` then the user has redirected stdout, so we shouldn't use interactive mode
+  * If the user's terminal doesn't support ANSI codes, then we shouldn't use interactive mode. Support for ANSI codes can be probed through environment variables as follows
+    * on Windows, if 'WT_SESSION' is non-null then Windows Terminal is being used, so we do support ANSI codes
+    * on all platforms, if the `TERM` variable is set to `dumb` then ANSI codes are not supported
+    * otherwise ANSI codes should be considered to be supported
+
+
+### Breaking out of the prompt loop
+
+CTRL/CMD+C should always allow the user to break out of the prompt loop.
+
+### Concerns/open questions
+
+* Many templates have symbols that are 'internal' and/or easily confused - e.g. TargetFrameworkOverride and Framework. How to coalesce these to prevent multiple prompts for something a user views as one item?
+  * For the first iteration, we should have a list of symbols to skip
+* Some parameters are not something I'd expect a user to ever want to change - e.g. skipRestore. Should we have another property marker that hides them?
+  * For initial version, prompt for these too. We can tweak logic based on feedback.
+* Which of the 'general' invocation options like -n and -o should we present here?
+* Should we have a new `prompt` property for this? Descriptions feel odd grammatically (which is why I didn't use them in the below example).
+  * Yes, and this new property should exist only in the `dotnetcli.host.json` file
+
+### Examples
+
+Here's what the C# Console template might look like using just the data we have available right now:
+
+![](sample-interactive-mode.png)
+
+Key points are the use of defaults where available, and the slight use of color to highlight property/symbol names vs the descriptions.
+One thing that can't been seen here is the use of choices - the 'framework' parameter showed a selection prompt and let me enter that.
+
+## References
+
+* [MSBuild LiveLogger](https://github.com/dotnet/msbuild/tree/main/src/MSBuild/FancyLogger)
```

Comment:
Please double check the link, I get a 404

Reply:
It has been recently renamed to LiveLogger, so the name is right on the doc but the link doesnt quite match rn.

Reply:
Thanks for the fix @maridematte - @donJoseLuis good to clear the change request?